### PR TITLE
feat: `v1/repos/:owner/:repo/:login/contributions`

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

Allow for fetching contributions of sole contributors for a single repo given an owner/repo pair for a repos "full_name" and the login of the user in question.

Example:

```
curl -X 'GET' \
  'http://localhost:3003/v1/repos/open-sauced/api/jpmcb/contributions' \
  -H 'accept: application/json'
```

returns

```
{
  "login": "jpmcb",
  "commits": 38,
  "prs_created": 19
}
```

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/api/issues/380

I decided to add a different endpoint instead of tacking this onto `v1/contributors/search` since that endpoint is already pretty overloaded and there wasn't a great way to do this efficiently with all the filters already on that query.

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by jpmcb** [comment]
Force pushed to add both `range` and `prev_days_start_date` to allow for dynamic time block windows:

```
curl -X 'GET' \
  'http://localhost:3003/v1/repos/open-sauced/api/jpmcb/contributions?prev_days_start_date=30&range=90' \
  -H 'accept: application/json'
```

```
{
  "login": "jpmcb",
  "commits": 9,
  "prs_created": 19
}
```

**Comment by jpmcb** [comment]
Force pushed to fix tests.

File: src/contribution/dtos/repo-user-contributions.dto.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    description: "Number of PRs reviewed by a user login",
+    example: 0,
+    type: "integer",
+  })
+  @Column({
+    type: "bigint",
+    select: false,
+    insert: false,
+  })
+  prs_reviewed: number;
```
Comment (nickytonline) [inline src/contribution/dtos/repo-user-contributions.dto.ts:51]: Are all these fields being created here and below to prepare for backfilling this data?

File: src/contribution/contribution.service.ts [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+import { DbRepoLoginContributions } from "./entities/repo-user-contributions.entity";
 
 @Injectable()
 export class ContributionService {
   constructor(
     @InjectRepository(DbContribution, "ApiConnection")
-    private contributionRepository: Repository<DbContribution>
+    private contributionRepository: Repository<DbContribution>,
+    @InjectRepository(DbUser, "ApiConnection")
+    private usersRepository: Repository<DbUser>
```
Comment (nickytonline) [inline src/contribution/contribution.service.ts:19]: nit (naming): I'm understanding this as a user's repo?

```suggestion
    private userRepository: Repository<DbUser>
```

**Review by nickytonline — APPROVED** [review]
Looks good, just a couple questions, nothing blocking.

🚢

File: src/contribution/dtos/repo-user-contributions.dto.ts (hunk 1 above)
Comment (jpmcb) [reply src/contribution/dtos/repo-user-contributions.dto.ts:51]: Essentially yes: it's a very similar DTO to the ones used in the user lists. I figured in the future we'll also want to surface those alongside `commits` and `pr_reviews`

File: src/contribution/contribution.service.ts (hunk 2 above)
Comment (jpmcb) [reply src/contribution/contribution.service.ts:19]: Good catch! 🙈

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**